instance/models/load_balancer.py

```diff
@@ -210,22 +264,28 @@ def reconfigure(self, triggering_instance_id=None):
         reconfiguration of the load balancer.
         """
         self.logger.info("Reconfiguring load-balancing server %s", self.domain)
-        with self._configuration_lock():
-            self.run_playbook(self.get_ansible_vars(triggering_instance_id))
+        self.run_playbook(self.get_ansible_vars(triggering_instance_id))
 
+    @modifies_configuration
     def deconfigure(self):
         """
         Remove the configuration fragment from the load-balancing server.
         """
+        self.logger.info("Deconfiguring load-balancing server %s", self.domain)
         fragment_name = settings.LOAD_BALANCER_FRAGMENT_NAME_PREFIX + self.fragment_name_postfix
-        with self._configuration_lock():
-            self.run_playbook(
-                "FRAGMENT_NAME: {fragment_name}\nREMOVE_FRAGMENT: True".format(fragment_name=fragment_name)
-            )
+        self.run_playbook(
+            "FRAGMENT_NAME: {fragment_name}\nREMOVE_FRAGMENT: True".format(fragment_name=fragment_name)
+        )
 
     def delete(self, *args, **kwargs):
         """
         Delete the LoadBalancingServer from the database.
         """
         self.deconfigure()
         super().delete(*args, **kwargs)
+
+    def _configuration_lock(self, **kwargs):
+        """
+        A Redis lock to protect reconfigurations of this load balancer instance.
+        """
+        return cache.lock("load_balancer_reconfigure:{}".format(self.domain), timeout=900, **kwargs)
```

Comment:
Should we make the timeout configurable?  We'll probably never need to change it, so I think it's fine.

Reply:
We can do that, sure. I don't like raw numbers in a codebase too much, so I'll add the var.

Reply:
P.S. I'm going to do this near the end of the PR's lifetime, to give myself time to do higher-priority things.